crates/uv/src/install_source.rs

```diff
@@ -0,0 +1,74 @@
+#![cfg(not(feature = "self-update"))]
+
+use std::{
+    ffi::OsStr,
+    path::{Path, PathBuf},
+};
+
+/// Known sources for uv installations.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub(crate) enum InstallSource {
+    Homebrew,
+}
+
+impl InstallSource {
+    /// Attempt to infer the install source for the given executable path.
+    fn from_path(path: &Path) -> Option<Self> {
+        let canonical = path.canonicalize().unwrap_or_else(|_| PathBuf::from(path));
+
+        let components = canonical
+            .components()
+            .map(|component| component.as_os_str().to_owned())
+            .collect::<Vec<_>>();
+
+        let cellar = OsStr::new("Cellar");
+        let formula = OsStr::new("uv");
+
+        if components
+            .windows(2)
+            .any(|window| window[0] == cellar && window[1] == formula)
+        {
+            return Some(Self::Homebrew);
+        }
+
+        None
+    }
+
+    /// Detect how uv was installed by inspecting the current executable path.
+    pub(crate) fn detect() -> Option<Self> {
+        Self::from_path(&std::env::current_exe().ok()?)
```

Comment:
Noting: I think this works because `current_exe` returns the resolved path on macOS, but this isn't actually guaranteed to be true:

https://doc.rust-lang.org/std/env/fn.current_exe.html#platform-specific-behavior

```
% which uv
/opt/homebrew/bin/uv
% readlink -f $(which uv)
/opt/homebrew/Cellar/uv/0.9.11/bin/uv
```

For full generality, we probably need `std::fs::read_link` or `std::fs::canonicalize` as an intermediate step here.